controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +571,122 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS records for deleted services/pods
```

Comment:
this PR makes the function very long (it was ~90 lines, now it's 200+). Let's consider yanking some code out of the function into helper functions to control complexity. E.g. this snippet to clean up DNS records could live in its own helper.

Reply:
Done